Src/libCZI/subblock_cache.cpp

```diff
@@ -90,6 +85,10 @@ void SubBlockCache::Prune(const PruneOptions& options)
 
 void SubBlockCache::PruneByMemoryUsageAndElementCount(std::uint64_t max_memory_usage, std::uint32_t max_element_count)
 {
+    // TODO(JBL): This is a very simple implementation of the prune operation. We determine the oldest element and remove it.
```

Comment:
Could the concern of iterating all over the elements be alleviated by structuring the cache impl more like an LRU cache in that we leverage a double-linked list as the underlying cache structure w/ a hashmap to provide quick get access?

More concretely: rather than keeping an LRU value, we move entry to head of dbl-linked list on access. Anything that is to be removed will be cut off from the tail based on size concerns (I confess we may still need to iterate a few elements here but at least we have an implicit order).